docs/cody/capabilities/agentic-chat.mdx

```diff
@@ -0,0 +1,78 @@
+
+Cody's agentic chat experience is an AI agent that can evaluate context and fetch any additional context (OpenCtx, terminal, etc.) by providing enhanced, context-aware chat capabilities. It extends Cody's functionality by proactively understanding your coding environment and gathering relevant information based on your requests before responding. These features help you get noticeably higher-quality responses.
+
+This agentic chat experience aims to reduce the learning curve associated with traditional coding assistants by minimizing users' need to provide context manually. It achieves this through agentic context retrieval, where the AI autonomously gathers and analyzes context before generating a response.
+
+## Capabilities of agentic chat
+
+The agentic chat experience leverages several key capabilities, including:
+
+- **Proactive context gathering**: Automatically gathers relevant context from your codebase, project structure, and current task
+- **Agentic context reflection**: Review the gathered context to ensure it is comprehensive and relevant to your query
+- **Iterative context improvement**: Performs multiple review loops to refine the context and ensure a thorough understanding
+- **Enhanced response accuracy**: Leverages comprehensive context to provide more accurate and relevant responses, reducing the risk of hallucinations
+
+## Enable agentic chat
+
+Pro users can find the agentic chat option in the LLM selector drop-down. Enterprise customers must opt-in to access this agentic chat feature.
+
+![agentic chat interface](https://storage.googleapis.com/sourcegraph-assets/Docs/agenti%20-chat.jpg)
+
+### Getting agentic chat access for Enterprise customers
+
+Currently, agentic chat uses Claude 3.5 Haiku for the reflection steps and Claude 3.5 Sonnet for the final response to provide a good balance between quality and latency. Your enterprise instance must have access to both Claude 3.5 Sonnet and Claude 3.5 Haiku to use agentic chat.
+
+Additionally, enterprise users need to upgrade their supported client (VS Code, JetBrains, and Visual Studio) to the latest version of the plugin by enabling the following feature flags on their Sourcegraph Instance:
+
+- `agentic-chat-experimental` to get access to the feature
+- `agentic-chat-cli-tool-experimental` to allow [terminal access](#terminal-commands)
+
+## What can agentic chat do?
+
+Agentic chat can help you with the following:
+
+### Tool Usage
+
+It has access to a suite of tools for retrieving relevant context. These tools include:
+
+- **Code Search**: Performs code searches
+- **Codebase File**: Retrieves the full content from a file in your codebase
+- **Terminal**: Executes shell commands in your terminal
+- **Chat Memory**: Allows the agent to create personal notes that can be used across chat sessions
```

Comment:
```suggestion
```

Let's not include Chat Memory as tool for now as it is still in rough shape and we have just identified a bug with it.